dashboard/test/controllers/data_docs_controller_test.rb

```diff
@@ -84,4 +97,28 @@ class DataDocsControllerTest < ActionController::TestCase
     get :edit, params: {key: 'unknown_key'}
     assert_response :not_found
   end
+
+  test 'data is passed to edit_all page' do
+    sign_in @levelbuilder
+
+    get :edit_all
+    assert_response :ok
+
+    show_data = css_select('script[data-datadocs]').first.attribute('data-datadocs').to_s
+    assert_equal DataDoc.all.order(:name).map(&:serialize).to_json, show_data
+  end
+
+  test 'data doc is deleted through destroy route' do
+    sign_in @levelbuilder
+    data_doc_to_delete = create :data_doc, key: 'doc_to_delete'
+
+    post :destroy, params: {
+      key: data_doc_to_delete.key
+    }
+    assert_response :ok
+
+    assert_raise ActiveRecord::RecordNotFound do
+      data_doc_to_delete.reload
+    end
+  end
```

Comment:
oh, I just realized that a unit test here would be a great way to test out what happens when you try to delete a nonexistent entry. I'm not sure this is super critical, so I'm also fine with you waiting until you have the frontend PR and testing that manually.

Reply:
will do! I'll add it now to keep the backend changes in this PR!